### PR TITLE
Rename and add backcompat requires for ChefFS dirs

### PR DESCRIPTION
**Pull request by thommay**
Policies, ACLs, and Client Keys directories are renamed to match the
other directories, and backwards compatible requires are added.
Ensure that they all properly inherit from Repository::Directory, too

### CODE REVIEW

**Comment by thommay** [comment]
@chef/client-core

File: lib-backcompat/chef/chef_fs/file_system/repository/file_system_root_dir.rb [hunk 1]
```diff
@@ -24,6 +24,7 @@ module FileSystem
       module Repository
         class FileSystemRootDir < FileSystemEntry
           def initialize(file_path)
+            Chef.log_deprecation "This class is deprecated."
```
Comment (danielsdeleo) [inline lib-backcompat/chef/chef_fs/file_system/repository/file_system_root_dir.rb]: Probably should include the class name so if you happen upon this in a log you have something to grep for.

**Comment by danielsdeleo** [comment]
:+1:

File: lib-backcompat/chef/chef_fs/file_system/repository/file_system_root_dir.rb (hunk 1 above)
Comment (thommay) [inline lib-backcompat/chef/chef_fs/file_system/repository/file_system_root_dir.rb]: @danielsdeleo heh, right. Done :)

**Comment by lamont-granquist** [comment]
👍

**Pull request merged**